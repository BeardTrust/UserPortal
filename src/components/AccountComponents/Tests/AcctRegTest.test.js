import userEvent from "@testing-library/user-event";
import axiosMock from "axios";
import TestRenderer from 'react-test-renderer'
import AccountRegistration from "../RegisterAccount/AccountRegistry";
import { act, render, screen } from '@testing-library/react';

jest.mock('axios')
describe('Account Registration Test', () => {

    const data = {
        data: {
            account: {
                userId: 'cf8bd72a-4b4c-42f1-8dc3-06cc2dc2cb8a',
                accountId: 'cf8bd72a-4b4c-42f1-8dc3-06cc2dc2cb8a',
                balance: '1000',
                interest: '1',
                nickname: 'name',
                type: 'Recovery',
                activeStatus: 'true',
                createDate: '2001-01-01'
            }
        }
    };

    it('Expects the snapshot to match the display', async () => {
        const tr = TestRenderer.create(<AccountRegistration/>);
        expect(tr.toJSON()).toMatchSnapshot();
    });

    it('Makes a post request with account data', async () => {
        const promise = Promise.resolve(data);
        axiosMock.post.mockResolvedValueOnce(promise);
        const singleAccount = render(<AccountRegistration url={'/test'} />);
        const regButton = screen.getByText('Register');
        userEvent.click(regButton);
        await act(() => promise);
        expect(axiosMock.post).toBeCalledTimes(1);
    });
})