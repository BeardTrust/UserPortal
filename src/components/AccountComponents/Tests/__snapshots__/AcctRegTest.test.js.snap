// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Registration Test Expects the snapshot to match the display 1`] = `
<section>
  <div>
    <form
      className="offset-4 col-3"
    >
      <h1
        className="text-center mt-5"
      >
        Set up a new BeardTrust Account today!
      </h1>
      <div>
        <label
          className="col-form-label form-label"
          htmlFor="username"
        >
          Give this account a nickname:
        </label>
        <input
          className="form-control"
          id="username"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          className="col-form-label form-label"
          htmlFor="username"
        >
          Input your initial deposit:
        </label>
        <input
          className="form-control"
          id="username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="mt-3 dropdown"
        required={true}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-success"
          data-toggle="dropdown"
          disabled={false}
          id="dropdown-basic"
          onClick={[Function]}
          type="button"
        >
          Select Account Type
        </button>
      </div>
      <div
        className="btn-group"
        role="group"
      >
        <button
          className="btn btn-primary mt-3 btn btn-primary"
          disabled={false}
          onClick={[Function]}
          title="registerButton"
          type="submit"
        >
          Register
        </button>
      </div>
    </form>
  </div>
</section>
`;
